pkg/aws/policies.go

```diff
@@ -889,10 +879,37 @@ func (c *awsClient) DeleteInlineRolePolicies(role string) error {
 	return nil
 }
 
+func (c *awsClient) isPolicyAttachedToEntity(policyArn string) (bool, error) {
+	policyOutput, err := c.iamClient.GetPolicy(&iam.GetPolicyInput{PolicyArn: aws.String(policyArn)})
+	if err != nil {
+		return false, err
+	}
+
+	isAttached := false
+	if *policyOutput.Policy.AttachmentCount > 0 {
+		isAttached = true
+	}
+
+	return isAttached, nil
+}
+
 func (c *awsClient) deletePolicies(policies []string) (*iam.DeletePolicyOutput, error) {
 	var output *iam.DeletePolicyOutput
-	var err error
+
 	for i := range policies {
```

Comment:
np, why not:
```suggestion
	for _, policy := range policies {
```

Reply:
Changed.

Reply:
ah, I guess that's why it was using `i`, sorry for the noise:
```
 pkg/aws/policies.go:912:76: G601: Implicit memory aliasing in for loop. (gosec)
		output, err = c.iamClient.DeletePolicy(&iam.DeletePolicyInput{PolicyArn: &policy})
		                                                                         ^
make: *** [lint] Error 1